data-model/src/main/java/io/micronaut/data/model/jpa/criteria/impl/AbstractPersistentEntityQuery.java

```diff
@@ -258,7 +258,11 @@ public Self forUpdate(boolean forUpdate) {
 
     @Override
     public Self where(Expression<Boolean> restriction) {
-        predicate = new ConjunctionPredicate(Collections.singleton((IExpression<Boolean>) restriction));
+        if (restriction instanceof ConjunctionPredicate conjunctionPredicate) {
```

Comment:
Merged from 4.9.x AbstractPersistentEntityCriteriaQuery into these separate classes that are created in 4.10.x
This maybe was not be needed for this fix https://github.com/micronaut-projects/micronaut-data/pull/3192 but looked like minor improvement